### PR TITLE
IN/FE: Full integration of Notifications

### LINKED ISSUE #10: BE: Setup Production MongoDB
\- Set up the MongoDB for the production database. Make sure the database is prepared for deployment this week.
\- Share the URI with your team on Slack when it's set up so they can setup the correct environment variables.

### PR DESCRIPTION
**Pull request by Habib-Ahmad**
### What this PR does (required):
-This PR uses socket.io to listen for incoming notifications and alert the user.
-All unread notifications can be accessed from the notifications button.
-When the notification drop down is opened, messages are marked as read.
-Clicking a notification takes the user to the appropriate route (i.e bookings page for bookings notifications, messages for user notifications)

### Screenshots / Videos (front-end only):
image

image

### Any information needed to test this feature (required):
- You need to be logged in. I used postman to generate new notifications as there is no way to do so in the UI yet

### CODE REVIEW

**Comment by sundayezeilo** [comment]
Hey @Habib-Ahmad, please confirm that you're using the appropriate base branch for this PR

**Comment by Habib-Ahmad** [comment]
> Hey @Habib-Ahmad, please confirm that you're using the appropriate base branch for this PR

This branch was created from main. Did you notice something wrong?

File: client/src/components/Navbar/Notification/Notifications.tsx [hunk 1]
```diff
@@ -0,0 +1,124 @@
...
+const Notification: React.FC<{
+  notif: INotification;
+  handleClose: () => void;
+}> = ({ notif, handleClose }) => {
+  const classes = useStyles();
+
+  const date = new Date(notif.date);
+  const [month, day, year] = [date.getMonth(), date.getDate(), date.getFullYear()];
+
+  const route = notif.type === 'booking' ? '/bookings' : notif.type === 'user' ? '/messages' : {};
```
Comment (moffatethan) [inline client/src/components/Navbar/Notification/Notifications.tsx:27]: A bit odd approach, ideally this data is attached on the notification itself but for times sake, this is fine for now.

**Review by moffatethan — CHANGES_REQUESTED** [review]
Can we see the UI of the notifications instead of just the circle?

**Review by moffatethan — APPROVED** [review]

**Pull request merged**